lib/light-service/organizer/reduce_when.rb

```diff
@@ -0,0 +1,19 @@
+module LightService
+  module Organizer
+    class ReduceWhen
+      extend ScopedReducable
+
+      def self.run(organizer, value_key, cases, else_steps)
+        lambda do |ctx|
+          return ctx if ctx.stop_processing?
+
+          steps = cases[ctx[value_key].to_sym] || else_steps
```

Comment:
As far as I understand, `ctx[value_key].to_sym` means that we support only symbols as possible values.

```ruby
TestReduceWhen.call(:number => 0, :incr_num => 2)
```

e.g., won't be supported. I'm imagining something like

```ruby
matched_case = cases.keys.find { |when| when.eql?(ctx[value_key]) }
steps = cases[matched_case] || else_steps
```

which should introduce support for a lot - if not all - value types, e.g:

```ruby
reduce_when(
          :current_value,
          {
            1 => [TestDoubles::AddsOneAction],
            [:ok, 'one'] => [TestDoubles::AddsTwoAction],
            :one => [TestDoubles::AddsThreeAction],
            FooObject => [TestDoubles::AddsFooAction]
          },
          :els => [TestDoubles::FailureAction]
        )
```

What do you think about the idea? Please consider I'm writing down the code w/o any sort of test/try.

I hope to not have misunderstood it all 🤞

Reply:
> n.d.r. maybe `when === ctx[value_key]` would be better that `.eql?` becouse it's the same comaprison method used by the `case` expression. But it hopefully it does not change the concept I was trying to deliver ;)